composer.json

```diff
@@ -26,7 +26,8 @@
         "guzzlehttp/psr7": "^1.4",
         "mockery/mockery": "^1.0",
         "php-http/message": "^1.6",
-        "php-http/mock-client": "^1.2"
+        "php-http/mock-client": "^1.2",
+        "phpunit/phpunit": "^8.2"
```

Comment:
Please do not require phpunit as a project dependency.

Reply:
How can we run the unit tests, without PHPUnit? It is a dev dependency so it does not get in the way. Also there is a dependency on Mockery as well, is it different somehow?

Just curious, of course if you want I'll remove it, but I'd like to understand the reasoning.

Reply:
You have different way to install PHPUnit: https://phpunit.de/getting-started/phpunit-8.html

If you want IDE completion, I would recommend to use the .phar.

Reasons to not require it:

1. It's a dependency hell, making the project setup heavier.
2. When you install a .phar, you are sure it's a well tested and packaged version with exact dependency. With composer, PHPUnit might crash because of an update of one of its depencies
3. On some project, you might have some dependencies conflict with packages you want to require and phpunit.

The alternative way would be to use the [composer-bin](https://packagist.org/packages/bamarni/composer-bin-plugin) plugin, but it will not solve reason `2`.

OFC, the reasons are not shared by every one and I'm not pretending to bring the real truth here. :wink:

Reply:
Got it, thanks for explaining :D